File: Backend/sporty/src/main/java/tn/esprit/sporty/Stats/model/Stats.java
package tn.esprit.sporty.Stats.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import tn.esprit.sporty.Authentification.model.User;

@Entity
@Data
public class Stats {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private int goalsScored;
    private int assists;
    private int minutesPlayed;
    private int interceptions;
    private int successfulPasses;
    @OneToOne
    @JoinColumn(name = "user_id", unique = true, nullable = false)
    @JsonIgnore
    private User user;
}

File: client/sportyfinalbackfront-master/src/app/models/stats.ts
export class Stats {
    id!: number;
    goalsScored!: number;
    assists!: number;
    minutesPlayed!: number;
    interceptions!: number;
    successfulPasses!: number;
    userId?: number;  // ID du joueur associé (optionnel)
  
    constructor(
      goalsScored: number,
      assists: number,
      minutesPlayed: number,
      interceptions: number,
      successfulPasses: number,
      userId?: number
    ) {
      this.goalsScored = goalsScored;
      this.assists = assists;
      this.minutesPlayed = minutesPlayed;
      this.interceptions = interceptions;
      this.successfulPasses = successfulPasses;
      this.userId = userId;
    }
  }
  